src/main/java/com/google/devtools/build/lib/analysis/BuildView.java

```diff
@@ -428,13 +462,6 @@ public AnalysisResult update(
       logger.atInfo().log(msg);
     }
 
-    Set<ConfiguredTarget> targetsToSkip =
-        new TopLevelConstraintSemantics(
-                skyframeExecutor.getPackageManager(),
-                input -> skyframeExecutor.getConfiguration(eventHandler, input),
-                eventHandler)
-            .checkTargetEnvironmentRestrictions(skyframeAnalysisResult.getConfiguredTargets());
-
```

Comment:
Do we still need to keep the original call for whoever's still relying on the legacy enforcement mechanism?

Reply:
Ah, good call. I jumped the gun a bit here. I'll restore this.

So the complete set of targets to skip is the union of `TopLevelConstraintSemantics.checkTargetEnvironmentRestrictions()` and the new functionality added in this patch. Am I understanding that correctly?